### ISSUE TITLE
Assembly 'Microsoft.Maui' Version=1.0.0.0 uses 'Microsoft.iOS, Version=16.0.0.0 which has a higher version than referenced assembly 'Microsoft.iOS' 15.4.300.0 -- FAILED.

### ISSUE THREAD

**Issue by VWilcox2000** [dotnet/maui#10626]
### Description

This is a recurrence of issue #8858 that was determined not to be a bug, but I am having difficulty finding a way to work around it with the latest visual studio releases.

With the latest preview of visual studio, it has the following workloads:

Installed Workload Ids      Manifest Version      Installation Source
---------------------------------------------------------------------
maui-windows                6.0.541/6.0.400       VS 17.3.32929.385
maui-maccatalyst            6.0.541/6.0.400       VS 17.3.32929.385
maccatalyst                 15.4.454/6.0.400      VS 17.3.32929.385
maui-ios                    6.0.541/6.0.400       VS 17.3.32929.385
ios                         15.4.454/6.0.400      VS 17.3.32929.385
maui-android                6.0.541/6.0.400       VS 17.3.32929.385
android                     32.0.476/6.0.400      VS 17.3.32929.385

The upgrade at least got rid of a version complaint blocking pairing about emscript, so I was quite hopeful.  I uninstalled xcode 14 on my mac and went back to 13.3 as that is supposed to be supported next week.  Also, with the preview version I now see iOS simulators again with 13.3 installed.

Building seems to go well, but I get the dreaded (by me in recent weeks at least) message above which does not provide much actionable information.  I already uninstalled all of visual studio and reinstalled and behavior remains consistent.

maui-check shows the following in red:

> .NET SDK - Workloads (6.0.301) Checkup...
  x android (Microsoft.NET.Sdk.Android.Manifest-6.0.300 : 32.0.440) not installed.
  x ios (Microsoft.NET.Sdk.iOS.Manifest-6.0.300 : 16.0.517) not installed.
  x maccatalyst (Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.400 : 15.4.465) not installed.
  x tvos (Microsoft.NET.Sdk.tvOS.Manifest-6.0.400 : 16.0.517) not installed.
  x macos (Microsoft.NET.Sdk.macOS.Manifest-6.0.400 : 12.3.465) not installed.
  x maui (Microsoft.NET.Sdk.Maui.Manifest-6.0.300 : 6.0.419) not installed.
! Managing Workload installation from the CLI is NOT recommended.  Instead you should install the latest Visual Studio
preview to automatically get the newest release of .NET MAUI workloads installed.
Are you sure you would like to continue the CLI workload installation? [y/n] (n):

But obviously, with latest preview and standard 2022 full installs, I have the latest.  Giving 'y' a shot, just gives the following:

Failed to update the advertising manifest microsoft.net.sdk.android: microsoft.net.sdk.android.manifest-6.0.300 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)"..
Failed to update the advertising manifest microsoft.net.sdk.ios: microsoft.net.sdk.ios.manifest-6.0.300 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)"..
Failed to update the advertising manifest microsoft.net.sdk.maccatalyst: microsoft.net.sdk.maccatalyst.manifest-6.0.300 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)"..
Failed to update the advertising manifest microsoft.net.sdk.macos: microsoft.net.sdk.macos.manifest-6.0.300 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)"..
Failed to update the advertising manifest microsoft.net.sdk.maui: microsoft.net.sdk.maui.manifest-6.0.300 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)"..
Failed to update the advertising manifest microsoft.net.sdk.tvos: microsoft.net.sdk.tvos.manifest-6.0.300 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)"..
Failed to update the advertising manifest microsoft.net.workload.emscripten: microsoft.net.workload.emscripten.manifest-6.0.300 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)"..
Failed to update the advertising manifest microsoft.net.workload.mono.toolchain: microsoft.net.workload.mono.toolchain.manifest-6.0.300 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)"..
Downloading microsoft.net.sdk.android.manifest-6.0.300.msi.x64 (32.0.440)
Workload installation failed. Rolling back installed packs...
Downloading microsoft.net.sdk.android.manifest-6.0.300.msi.x64 (32.0.301)
Installation rollback failed: One or more errors occurred. (microsoft.net.sdk.android.manifest-6.0.300.msi.x64::32.0.301 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)".)
Workload installation failed: One or more errors occurred. (microsoft.net.sdk.android.manifest-6.0.300.msi.x64::32.0.440 is not found in NuGet feeds C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\$(PACKAGE_SOURCE_NUGET_ORG)".)
Fix failed - Workload Install failed: `dotnet workload install --from-rollback-file
"C:\Users\User\AppData\Local\Temp\maui-check-2626c1fc\workload.json" android ios maccatalyst tvos macos maui --source
"$(PACKAGE_SOURCE_NUGET_ORG)"`

which rolls everything back so no changes are made.

Is this something where you can point me to something to change or is it likely I just need to wait for next week's build?

Thanks!
Victor

### Steps to Reproduce

MAUI project targeting net6.0-ios.

Visual Studio 2022 preview 3.0

Pair to mac succeeds.

Build almost succeeds but at the very end spits out:

CSC : error CS1705: Assembly 'Microsoft.Maui' with identity 'Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.iOS, Version=16.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065' which has a higher version than referenced assembly 'Microsoft.iOS' with identity 'Microsoft.iOS, Version=15.4.300.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065'

The app works fine on Android, but to do so, I have to remove iOS targeting for build to succeed.

my global.json:
{
  "sdk": {
    "version": "6.0.401"
  }
}

### Link to public reproduction project repository

n/a

### Version with bug

6.0 Release Candidate 2 or older

### Last version that worked well

6.0.400

### Affected platforms

iOS

### Affected platform versions

iOS

### Did you find any workaround?

I was able to build and deploy a single time immediately after rebuilding mac / windows machines entirely.  I think 2022 worked, but once I used preview it changed things up, but I was previously getting a version mismatch on Emscript until today's patch after which I'm getting the maui assembly version info message

### Relevant log output

```shell
Build started...
1>------ Build started: Project: YourCompany, Configuration: Debug Any CPU ------
1>Executing SayHello Task to establish a connection to a Remote Server. 
1>			Properties: 
1>				SessionId=83d9ddd4ba76b43fd4e51a150853625f3e4e2a02b9c8e8c1252149b606d298e2, 
1>				Addresss=10.67.42.107, 
1>				SshPort=22, 
1>				TcpPort=50660, 
1>				User=VWilcox, 
1>				AppName=YourCompany,
1>				VisualStudioProcessId=44476,
1>				ContinueOnDisconnected=False
1>C:\src\YourCompanyPro\Mobile\MAUI\YourCompany\YourCompanyState.Tasks.cs(21,49,21,73): warning CS0067: The event 'YourCompanyState.workflowTaskValueUpdated' is never used
1>C:\src\YourCompanyPro\Mobile\MAUI\YourCompany\YourCompanyState.Tasks.cs(21,49,21,73): warning CS0067: The event 'YourCompanyState.workflowTaskValueUpdated' is never used
1>Detected signing identity:
1>  Bundle Id: com.yourCompanyinc.yourCompany
1>  App Id: com.yourCompanyinc.yourCompany
1>C:\src\YourCompanyPro\Mobile\MAUI\YourCompany\YourCompanyState.Tasks.cs(21,49,21,73): warning CS0067: The event 'YourCompanyState.workflowTaskValueUpdated' is never used
1>CSC : error CS1705: Assembly 'Microsoft.Maui' with identity 'Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.iOS, Version=16.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065' which has a higher version than referenced assembly 'Microsoft.iOS' with identity 'Microsoft.iOS, Version=15.4.300.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065'
1>Done building project "YourCompany.csproj" -- FAILED.
1>YourCompany -> C:\src\YourCompanyPro\Mobile\MAUI\YourCompany\bin\Debug\net6.0-maccatalyst\maccatalyst-x64\YourCompany.dll
1>YourCompany -> C:\src\YourCompanyPro\Mobile\MAUI\YourCompany\bin\Debug\net6.0-windows10.0.19041.0\win10-x64\YourCompany.dll

Error List:
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1705	Assembly 'Microsoft.Maui' with identity 'Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.iOS, Version=16.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065' which has a higher version than referenced assembly 'Microsoft.iOS' with identity 'Microsoft.iOS, Version=15.4.300.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065'	YourCompany (net6.0-ios)	C:\src\YourCompanyPro\Mobile\MAUI\YourCompany\CSC	1	Active
```

**Comment by neoayi** [comment]
for me, uninstall all dotnet sdk, reinstall SDK 6.0, and VS 2022 17.3, it's OK.

**Comment by elGuille-info** [comment]
@neoayi I uninstall Visual Studio then re-install, I also need to uninstall SDK? I thought it was installed by VS.

**Comment by belmonmi** [comment]
@neoayi I'm glad your approach worked for you, but I'm sure this is not acceptable "solution" for most of developers with their development environments contains third party extensions that had to be re-installed and re-configured. Microsoft needs to fix the issue that they created with this VS update. 
Unfortunately, lately I cannot trust VS "stable" updates, after each update I only hope that I will be able to continue to work on my projects, especially when you have a deadline pressure hope transforms to fear.

**Comment by neoayi** [comment]
Actually, I just gave a plan. I don't like the whole development environment being unavailable because of one update. In fact, we have to make a certain judgment ourselves, that is, the preview version will definitely have problems. It's just that our Principal Product Manager didn't tell us the details, i.e., if you install multiple environments, you create a messy development environment. Actually, this has been bugging me for a day and I have spent a day solving this problem and trying to understand it as well.

**Comment by VWilcox2000** [comment]
> 
@neoayi Thanks for that information!  I've done that a few times in last couple of weeks, but I also install preview -- maybe that's breaking it again.   If I have time, I will experiment to confirm on my end.  Regards.

**Comment by VWilcox2000** [comment]
@belmonmi Thanks for that comment! -- it's reassuring that I'm not the only MAUI developer that, of late, can rarely get an app built let alone published for iOS.  You've helped me maintain my sanity.

**Comment by elGuille-info** [comment]
I uninstall the SDKs from the Visual Studio installer, re-install (previously try with dotnet workload restore on one of the projects) and the same... nothing to do with the iOS part of the project! 🙄

**Comment by hamedhajiloo** [comment]
Same issue!

**Comment by foximoxi** [comment]
Something is broken within the development process for MAUI and VS2022.
Last SDK 6.0.401 had problems to build and install workloads, now this. I've the same issue.

**Comment by belmonmi** [comment]
@foximoxi I think you nailed it. Sometimes mistakes happens and it is OK, sometime third-party products that you relay on change their interfaces/protocols (xCode 13 vs. xCode 14) and this is understandable as well. But when you have 2 or 3 release in the row that cripple or completely stop development - this is a sign. Clearly .NET MAUI iOS team need to take a good look at their development process and figure out why this keep happening. Before VS ver. 17.3.6 release, iOS was broken, but at least I was able to build and debug on Android and moving on, expecting that by the time I'm ready to release MS will figure out the iOS issues. But now, I cannot even build. :(

**Comment by VWilcox2000** [comment]
@belmonmi 

I at least have had luck working on Android by editing the project and removing iOS from the target frameworks:

          <Project Sdk="Microsoft.NET.Sdk">
          
          	<PropertyGroup>
                     <TargetFrameworks>net6.0-android</TargetFrameworks>
                     

You can achieve the same turning off the targets in the project, of course.

**Comment by hootanht** [comment]
> Same issue!

:)

**Comment by elGuille-info** [comment]
@VWilcox2000 Really you don't need to take out the iOS setting in the `TargetFramework` tag, and works in the others you have set. I'm testing applications with Android, iOS and Windows and only can't compile/deploy if I set iOS.

**Comment by belmonmi** [comment]
@VWilcox2000 
Thank you for the work around. RESPECT !

**Comment by VWilcox2000** [comment]
For some reason, my work visual studio account will not log into the report feedback page.  If someone else interested wouldn't mind posting it and linking here?  I'm hoping it's fixed by Tuesday, but it would be great to have an official issue ready just in case.

The good news is, I saw a new Xamarion.iOS 16.0 pushed to mac on pairing, suggesting MS /Maui team are actively working on it as they said in last patch notes.  This still does not help with the build issue on visual studio, but HOPEFULLY that's what we get on Tuesday.

**Comment by VWilcox2000** [comment]
> @VWilcox2000 Really you don't need to take out the iOS setting in the `TargetFramework` tag, and works in the others you have set. I'm testing applications with Android, iOS and Windows and only can't compile/deploy if I set iOS.

@elGuille-info that is QUITE interesting.  I've tried new projects, manually rebuilding a very simple csproj, etc, and for me, if iOS is in the targets, it throws that build error on all targets.  Could be related to the workflows... I do hope they come out with a comprehensive workflow manager with a UI sometime down the road.  You can delete the workflow folder and reinstall, but it's still a bit of a black box.

**Comment by belmonmi** [comment]
@VWilcox2000 
I think (conspiracy theory on my end :) ), that iOS MAUI team spend majority of their time on Mac not on PC. This is why their workload deployments works for them (High Five everyone, another successful release!), but no one is spending time to test it on PC.  This is my evidence:

1. Update your Visual Studio on Mac
2. run: dotnet new maui
3. open the project file and leave only iOS target (thanks again @VWilcox2000)
4. run: dotnet build
5. no errors - success 

Following is output from my console on Mac:

$ dotnet build
MSBuild version 17.3.2+561848881 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  Detected signing identity:
          
    Bundle Id: com.companyname._1
    App Id: com.companyname._1
  1 -> /Users/m*****/Documents/Temp/1/bin/Debug/net6.0-ios/iossimulator-x64/1.dll
  Optimizing assemblies for size, which may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.16

maui-check output looks very cheerful as well:

.NET SDK - Workloads (6.0.301) Checkup...
  ✔ android (Microsoft.NET.Sdk.Android.Manifest-6.0.300 : 32.0.440) installed.
  ✔ ios (Microsoft.NET.Sdk.iOS.Manifest-6.0.300 : 16.0.517) installed.
  ✔ maccatalyst (Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.400 : 15.4.465) installed.
  ✔ tvos (Microsoft.NET.Sdk.tvOS.Manifest-6.0.400 : 16.0.517) installed.
  ✔ macos (Microsoft.NET.Sdk.macOS.Manifest-6.0.400 : 12.3.465) installed.
  ✔ maui (Microsoft.NET.Sdk.Maui.Manifest-6.0.300 : 6.0.419) installed.
 
Connect the dots... ¯\_(ツ)_/¯

**Comment by pulmuone** [comment]
Same issue!

**Comment by jk-ltc** [comment]
For the record, this issue was reported to VS Feedback [here](https://developercommunity.visualstudio.com/t/CSC-:-CS1705---MicrosoftMaui-uses-Micr/10170928?space=8&ftype=problem&preview2=true&q=+which+has+a+higher+version+than+referenced+assembly+%27Microsoft.iOS%27), but that item is marked closed as a duplicate of this issue

**Comment by wdshh** [comment]
same issue!!

**Comment by VWilcox2000** [comment]
@FM1973 I completely agree... a couple of years ago, I just published to archives, then store, put in app password and done.  If not for Transporter, I wouldn't have anything in app store more recent than almost 2 years old, and much of the time I can't use the show IPA option, so I get to go on scavenger hunts.  I'm hopeful this is all growing pains as we transition from Xamarin into MAUI... exacerbated by xcode 14/iOS 16 introducing a large number of breaking changes.  I'm starting to get pressure from the rest of the company to rewrite our app in Swift, now...  :(

**Comment by yzbai** [comment]
same here

**Comment by AndreasReitberger** [comment]
I ran into the same issue.
I tried a lot, nothing was working. I guess it's messed up by multiple installed Visual Studio Versions.
In my case, `VS 2019 Enterprise`, `VS 2022 Community` and `VS 2022 Com. Preview`.

While `Xamarin.Forms` is working, I stuck with the same error on `MAUI` as mentioned by the OP. So I decided to reinstall my Windows 10, to start with a completely fresh system to test.
Afterwards I just installed VS 2022 (which should be working for MAUI and XCode 14).

I guess re-installing Windows is not applicable for the most to fix this. In my case, it did the job and was worth the efforts.
Maybe, if someone knows how to completely remove the VS stuff from your computer (included all dotnet sdks and so on), this might be sufficient.

**Comment by elGuille-info** [comment]
I managed to get it to work on another computer.
There I have installed the Visual Studio 2022 Preview (Microsoft Visual Studio Community 2022 (64-bit) - Preview - Version 17.4.0 Preview 3.0) and I have had no problems with the workloads, or with anything... I have tried with iOS in a iPhone 7 plus device (I don't have access to a Mac) and with Windows, with Android I haven't tested, but I guess it will work.
The installed .NET version is: 7.0.100-rc.2.22477.23 and the .NET MAUI version is: 7.0.100-rc.1.

Guillermo

**Comment by belmonmi** [comment]
@elGuille-info thank you for your efforts. It looks like VS Preview code/workloads got trickled in to PROD/Release version of VS. This is again reinforced @foximoxi point, that deployment process for .NET MAUI is broken.

**Comment by elGuille-info** [comment]
@belmonmi yes, but now I can try on iOS devices 😉

**Comment by yzbai** [comment]
This saved me.
https://github.com/dotnet/sdk/issues/27587#issuecomment-1279760064

**Comment by jayhayman-hdd** [comment]
For reference, I only have one stable version of VS 2022 installed (Enterprise 17.3.6), no preview has every touched this dev box. Using Xcode 13.4, **and I get this issue too**. 

Only happened since the last VS update. VS 2022 Mac is all up to date too. Xcode 14 is installed on the mac but the path is pointing at 13.3, since VS iOS agent is still telling my Xcode 14 is not yet supported. I have uninstalled and reinstalled SDKs on both platforms but with no success.

Good job my customers are not yet relying on MAUI to deliver updates to our app otherwise we'd be in serious trouble., we purposely kept our Xamarin.Forms version running in parallel.

I get that new platforms have issues; been there before, still carrying the scars, and I am willing to wait for things to get better with MAUI as the potential is amazing, But, I can't justify the effort it takes to "maintain" a solid build process with MAIU yet as it's causing more support issues with every update.

**Comment by SoaSoftNet** [comment]
问题已经解决了.  dotnet workload update   .17.3.6已测试，OK

**Comment by jk-ltc** [comment]
Same. Still can't build a MAUI app with iOS targeting. No workarounds.

**Comment by jk-ltc** [comment]
@belmonmi - Brilliant. Got me over the last hurdle. 
Seems like the `dotnet workload install maui` had a significant effect (there was basically nothing in my metadata\workloads folder prior to running that install command).
The steps I took previously that had made progress, but not completely solved it were to back up and delete the contents of:
%UserProfile%\.dotnet\sdk-advertising
C:\ProgramData\dotnet\workloads
C:\Program Files\dotnet\packs
C:\Program Files\dotnet\sdk-manifests
and following that, running a repair from Visual Studio Installer.
Thanks @belmonmi - I am back to being able to build and run on iOS simulator.

**Comment by VWilcox2000** [comment]
After the last couple of updates, I committed to .net 7.0 rc2 and was able to build.  On my paired local mac, it stopped working at some point and I went back to getting a black screen after the splash on every run -- with the same code.  Also, then preview started doing the dreaded stop midway loading projects at the same point every time -- the "fixed" issue that happens every once in a while.

      Frustrated with the 5% up-time trying to work on iOS version of my app, I finally decided to try MacInCloud.com and chose the $25 option + $10 for SSH connectivity for one of the fastest mac's built.  It comes with xcode 14.01 already installed.  After that, my app which had been black-screening built and RAN.  There is also a MacStadium option which might be better option for some users depending on the plans.  The remote in software from Windows is easy to use also.  Why this worked instead of my local machine which I wiped and reinstalled on?  Don't know.  Inconsistent files left around from a previous version breaking the current version?  No idea and the generally non-informative (or worse, misguiding) messages of all xamarin tools don't help, but if you are as desperate as I was -- it seems to be working well for me... and if the machine ends up in a bad state, you can wipe and start with the initial image again.
